### ISSUE TITLE
Ability to Create additional Static MGMT Route with DHCP enabled

### ISSUE THREAD

**Issue by f5-applebaum** [F5Networks/f5-declarative-onboarding#226]
In specific cases, ex. Where routes are not advertised in DHCP lease,  i need to create static mgmt routes to supplement. 

ex. Adding a metadata route
```
[auser@f5vm010:Active:In Sync] ~ # tmsh list sys management-route
sys management-route dhclient_route1 {
    gateway 10.3.0.1
    network 168.63.129.16/32
}
sys management-route dhclient_route2 {
    gateway 10.3.0.1
    network default
}
sys management-route azureMetadata {
    gateway 10.3.0.1
    network 169.254.169.254/32
}

```

However, when mgmt routes are added to DO, DHCP is disabled entire. 

ex.
https://github.com/F5Networks/f5-declarative-onboarding/issues/185
 
I would like to keep my Dyanamic IP/Routes from DHCP and add a static route in my DO declaration.

Static Routes should be able to live along side Dynamic information (ex. just because you run dynamic routing protocol like ospf, doesn't mean can't have static routes).

**Comment by dstokesf5** [comment]
Do you expect to declare the DHCP routes, or are you expecting DO to leave anything with a description of "configured-by-dhcp" alone?

**Comment by f5-applebaum** [comment]
Hi, 

In the original use case, any routes declared in DO would strictly be "additive". 

ex.  
`
[
   {
      "extensionType": "do",
      "type": "inline",
      "value": {
         "schemaVersion": "1.0.0",
         "class": "Device",
         "label": "DHCP Routes and Static Management Route",
         "async": true,
         "Common": {
            "class": "Tenant",
            "dbVars": {
               "class": "DbVariables",
               "provision.extramb": 1000,
               "restjavad.useextramb": true,
               "ui.advisory.color": "blue",
               "ui.advisory.text": "Provisioned via DO"
            },
            "myDns": {
               "class": "DNS",
               "nameServers": [
                  "168.63.129.16"
               ]
            },
            "myNtp": {
               "class": "NTP",
               "servers": [
                  "pool.ntp.org"
               ],
               "timezone": "UTC"
            },
            "myProvisioning": {
               "asm": "nominal",
               "class": "Provision",
               "ltm": "nominal"
            },
            "mySystem": {
               "class": "System",
               "autoPhonehome": true,
               "hostname": "{{{HOST_NAME}}}.local"
            },
            "azureMetadata": {
                "class": "ManagementRoute",
                 "gw": "10.0.0.1",
                 "network": "169.254.169.254/32"
            }
         }
      }
   }
]

Where DHCP would provide the typical MGMT IP, MGMT Routes and Default Route and any routes with "description configured-by-dhcp" would be left alone. 

ex.
[admin@mybigip-vmss-0:Active:Standalone] ~ # tmsh list sys management-route
sys management-route dhclient_route1 {
    description configured-by-dhcp
    gateway 10.0.0.1
    network 168.63.129.16/32
}
sys management-route default {
    description configured-by-dhcp
    gateway 10.0.0.1
    network default
}

You would always need to leave those alone as if you try to statically create static routes that conflicted with DHCP installed routes, DHCP would actually error out on the first conflict and not install other routes. 

However, good point. If a user accidentally provided routes that conflicted with routes that DHCP already installed like you said, 

 [
   {
      "extensionType": "do",
      "type": "inline",
      "value": {
         "schemaVersion": "1.0.0",
         "class": "Device",
         "label": "DHCP Routes and Static Management Route",
         "async": true,
         "Common": {
            "class": "Tenant",
            "dbVars": {
               "class": "DbVariables",
               "provision.extramb": 1000,
               "restjavad.useextramb": true,
               "ui.advisory.color": "blue",
               "ui.advisory.text": "Provisioned via DO"
            },
            "myDns": {
               "class": "DNS",
               "nameServers": [
                  "168.63.129.16"
               ]
            },
            "myNtp": {
               "class": "NTP",
               "servers": [
                  "pool.ntp.org"
               ],
               "timezone": "UTC"
            },
            "myProvisioning": {
               "asm": "nominal",
               "class": "Provision",
               "ltm": "nominal"
            },
            "mySystem": {
               "class": "System",
               "autoPhonehome": true,
               "hostname": "{{{HOST_NAME}}}.local"
            },
            "azureMetadata": {
                "class": "ManagementRoute",
                 "gw": "10.0.0.1",
                 "network": "169.254.169.254/32"
            },
	   "dhclient_route1": {
	       "class": "ManagementRoute",
	       "gw": "10.0.0.1",
	       "network": "168.63.129.16/32"
	    },
	    "default": {
	        "class": "ManagementRoute",
	        "gw": "10.0.0.1",
	        "network": "default"
	   }
         }
      }
   }
]

You could 
1) Be strict and error out that there is a conflict with existing DHCP routes. "dhclient.mgmt: disable" required. 
or 
2) Still succeed (just leave those routes alone) and give a "Warning: routes 0.0.0.0 and 168.63.129.16/32 are already configured via DHCP". Success is still justified as the desired endstate that 3 routes will be configured (2 by DHCP and one static DO did install). 

If had to chose, would prefer the later as allows customer have the configuration reflect more of what's desired on the instance, vs. just piece of the picture. However, the requirement is just that static routes can be provisioned along side of dynamic DHCP routes.

**Comment by dstokesf5** [comment]
Thank you for the thorough response.

**Comment by dstokesf5** [comment]
This has been added to our internal product backlog as AUTOTOOL-2491.

**Comment by seattlevine** [comment]
In a [similar issue](#193), we are adding a flag `preserveOrigAuthKeys` to indicate that we should keep both the original (first time DO is run) keys and the keys in the declaration. We'd like to do the same here with a flag `preserveOrigDhcpRoutes`. Is that acceptable?

**Comment by f5-applebaum** [comment]
sure. Where would that be defined. Example if I created 2+ custom mgmt routes (in addition to the ones already provisioned via DHCP. Assuming in *each* ManagementRoute definition just in case there is a conflict with a DHCP provided route?

**Comment by megamattzilla** [comment]
I'm working with a customer who has experienced this issue as well. For now, we workaround by avoiding adding management routes, but it would be great to see this issue fixed so we could add mgmt routes.

**Comment by dstokesf5** [comment]
In version 1.23.0 you will be able to specify a property to preserve the dhcp routes:
```json
{
    "class": "Device",
    "async": true,
    "schemaVersion": "1.23.0",
    "Common": {
        "class": "Tenant",
        "system": {
            "class": "System",
            "preserveOrigDhcpRoutes": true
        },
        "newManagementRoute": {
            "class": "ManagementRoute",
            "network": "1.2.3.4",
            "gw": "4.3.2.1"
        }
    }
}
```

**Comment by dstokesf5** [comment]
This issue has been resolved with the 1.23.0 release of Declarative Onboarding.
